web/components/App.jsx

```diff
@@ -0,0 +1,215 @@
+      const duplicates = _.filter(self.slice(ZERO, index), ['id', value.id]);
+      return duplicates.length === ZERO ? value : false;
+    }).sort((a, b) => moment(a.createdAt).diff(moment(b.createdAt)));
+
+    setEvents(allEvents);
+
+    if (container && container.scrollHeight - container.scrollTop - DFB <= container.clientHeight) {
+      setScroll(true);
+    } else {
+      const latestEvent = newEvents.length && newEvents[ZERO];
+      setToast(latestEvent && latestEvent.context &&
+        (filter === 'All' || filter.includes(latestEvent.context.type)));
+    }
+  }
+
+  useEffect(() => {
+    const previousEvents = events;
+    const latestEvent = response.length && response[ZERO];
+
+    if (scroll && container) {
+      container.scrollTop = container.scrollHeight;
+      setScroll(false);
+    }
+
+    if (latestEvent && !_.find(previousEvents, latestEvent)) {
+      handleNewEventsReceived(previousEvents, response);
+    }
+  });
+
+  /**
+   * changes the active message filter
+   * @param {object} type - filter type that was selected
+   */
+  function changeFilterType(type) {
+    if (filter === 'All' || type === 'All') {
+      setFilter(type);
+    } else if (filter === type) {
+      setFilter('All');
+    } else if (filter.includes(type)) {
+      setFilter(filter.replace(type, ''));
+    } else {
+      setFilter(filter + type);
+    }
+
+    props.getEventsByType(type);
+    setScroll(true);
+  }
+
+  /**
+   * sends a chat message.
+   */
+  async function sendChat() {
+    if (currentText === '') return;
+    setPendingMessage(true);
+
+    try {
+      await messageService.sendMessage(currentText, props.user);
+      setCurrentText('');
+      setPendingMessage(false);
+    } catch (err){
+      console.error(`message failed to send - ${err.message}`);
+    }
+  }
+
+  /**
+   * @param {object} e - event object from file dropped on component
+   */
+  async function uploadAttachment(e) {
+    setFileDraggedOver(false);
+    e.stopPropagation();
+    e.preventDefault();
+
+    try {
+      const data = e.dataTransfer ? e.dataTransfer : e.target;
+      const { files } = data;
+      await attachmentService.postAttachment(files[0], props.selectedChatter);
+    } catch (err){
+      console.error(`failed to upload file - ${err.message}`);
+    }
+  }
+
+  /**
+   * @param {object} e - event object on file dragged over component
+   */
+  function fileDraggedOverHandler(e) {
+    setFileDraggedOver(true);
+    e.stopPropagation();
+    e.preventDefault();
+  }
+
+  return (
+    <div>
+      <FilterHeader filter={filter} changeType={changeFilterType} />
+      <div
+        id="file-drop"
+        onDragOver={fileDraggedOverHandler}
+        className={ fileDraggedOver ? 'file-dragged-over' : 'no-file-dragged-over' }
+        onDragLeave={() => setFileDraggedOver(false)}
+        onDrop={uploadAttachment}
+      >
+        <ul
+          className="slds-chat-list slds-m-bottom--xx-small"
```

Comment:
Don't know if this an issue in react, but I think slds suggests to use "_" instead of "--" so it doesn't clash with html comments. So you could write "slds-m-bottom_xx-small"